File: medium/python/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        longest = 0
        long = 0
        queue = []
        for char in s:
            if char in queue:
                index = queue.index(char)
                queue[:index + 1] = []
                longest = max(long, longest)
                long -= index
                queue.append(char)
            else:
                queue.append(char)
                long += 1
        return max(long, longest)

File: medium/golang/3-longest-substring-without-repeating-characters.go
package golang

func lengthOfLongestSubstring(s string) int {
	n := len(s)
	if n == 0 {
		return 0
	}
	left := 0
	right := 1
	length := 1
	charIndexes := make(map[uint8]int)
	charIndexes[s[left]] = left
	for right < n {
		if index, ok := charIndexes[s[right]]; ok && index >= left {
			left = index + 1
		}
		charIndexes[s[right]] = right
		length = max(length, right-left+1)
		right++
	}
	return length
}
